### ISSUE TITLE
Make `interaction_line_plot` 

### ISSUE THREAD

**Issue by mark-andrews** [mark-andrews/psyntur#18]
```r
# assumes tidyverse is loaded
interaction_line_plot <- function(y, x, by, data, ylim = NULL){
  
  Df <- group_by(data, {{x}}, {{by}}) %>% 
    summarise(mean = mean({{y}}),
              sem = sd({{y}})/sqrt(length({{y}})),
              .groups = 'drop')
  
  the_aes <- aes(x = {{ x }}, y = mean, group = {{ by }}, colour = {{ by }})

  p1 <- ggplot(Df, mapping = the_aes) + geom_point() + geom_line()
  
  p1 <- p1 + geom_linerange(aes(x = {{ x }}, ymin = mean - sem, ymax = mean + sem), size = .3)
  
  if (!is.null(ylim)){
    p1 <- p1 + scale_y_continuous(limits = ylim)
  }

  p1 + theme_classic() + scale_colour_brewer(palette = "Set1") 
  
}
```

E.g. 

```r
selfesteem2 <- read_csv('http://data.ntupsychology.net/selfesteem2.csv')
interaction_line_plot(y = score, x = time, by = treatment, data = selfesteem2, ylim = c(70, 100))
```

**Comment by jensroes** [comment]
I'm not actually sure why so many people use 1 * SEs for these kind of plots. I feel that they understate the variability in the data. Can we use CIs for that as  in
ggplot(data, aes(x = time, y = score, color = treatment)) +
  stat_summary(fun = mean, geom = "point") +
  stat_summary(fun.data = mean_cl_normal, geom = "errorbar", width = .1)

**Comment by jensroes** [comment]
I'd also suggest we should to horizontally position_dodge lines and points so there is no overlap.

**Comment by mark-andrews** [comment]
Your suggestion is better. But there is a bigger question, why use these types of plots at all? For example, a Tukey boxplot with jitter 
```r
library(tidyverse)
library(psyntur)

selfesteem2 <- read_csv('http://data.ntupsychology.net/selfesteem2.csv')
tukeyboxplot(y = score, x = time, by = treatment, jitter = T, data = selfesteem2)
```
will show us the central tendency and spread of all six groups. If we want to join up the three box of each of the two `treatment` types with lines, we could do that too.
By contrast, the line and error bars hides a lot of information. 

As I see it, these are "legacy" plots like barplots with error bars. People do them because others do them and everyone thinks they are expected and somehow right. People think that 1se on either side of mean is what they should do, or else they think that is what is expected of them.

My attitude is that we should be either plotting the data or we are plotting the results of a model. With the latter, for example, we could plot estimates and standard errors. This interaction_plot is not of a model's estimates and standard errors, it is from a crude means and standard deviations divided by sample sizes from the data for each group, and these are not the standard errors from the model. If we are plotting the data, we are better off with the Tukey boxplot etc. 

Basically then, I think the best option would be to use the model itself, possibly using `emmeans` or something and plot the estimates and confidence intervals *from the model*. That way, we are visualizing the model, which is then complementary to the visualization of the data with the boxplot.

**Comment by jensroes** [comment]
Absolutely agreed on all counts.

**Comment by mark-andrews** [comment]
Even though an interaction line plot with standard error bars is a vestige of a bygone era (when people thought barplots with standard errors were all you ever needed when pltting), I will add this function to the package just because we are using the raw ggplot in a workshop, and at the very least we should replace that.
